pkg/http/http_response_encoder.go

```
@@ -0,0 +1,61 @@
+}
+
+// sendError sends an error JSON response
+func (e *JSONEncoder) SendError(w http.ResponseWriter, statusCode int, errorMsg string) {
+	sendJson(w, statusCode, Response{Success: false, Error: errorMsg})
+}
+
+// sendJson is a helper function to send JSON responses
+func sendJson(w http.ResponseWriter, statusCode int, response Response) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(response)
```

golangci-lint (1.64.8)

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)